### PR TITLE
Issue45/admin cards

### PR DESCRIPTION
**Pull request by blatinier**

### CODE REVIEW

File: src/static/actions/cards.js [hunk 1]
```diff
@@ -9,6 +9,15 @@ import { USER_CARDS_FETCH_REQUEST,
...
+    CARD_FETCH_FAILURE,
+    CARD_DELETE_FAILURE,
+    CARD_DELETE_REQUEST,
+    CARD_DELETE_SUCCESS,
+    CARD_EDIT_FAILURE,
+    CARD_EDIT_REQUEST,
+    CARD_EDIT_SUCCESS,
     CARD_ADD_FAILURE,
     CARD_ADD_REQUEST,
     CARD_ADD_SUCCESS } from '../constants';
```
Comment (Far3t) [inline src/static/actions/cards.js]: Maybe use a:
import * as constants from '../constants';
and use it as constants.CARD_EDIT_FAILURE for example.
It's up to you.

File: src/static/actions/cards.js [hunk 2]
```diff
@@ -173,3 +182,162 @@ export function createCard(token, values) {
...
+        type: CARD_EDIT_REQUEST
+    };
+}
+
+export function cardEditSuccess() {
+    return {
+        type: CARD_EDIT_SUCCESS
+    };
+}
+
```
Comment (Far3t) [inline src/static/actions/cards.js]: May be lighter with arrowed functions

File: src/static/containers/AdminCardAdd/index.js [hunk 3]
```diff
@@ -57,59 +56,133 @@ class AdminCardAddView extends Component {
...
         },
     };
 
+    componentWillMount = () => {
+        const { slug } = this.props.match.params;
+        const { token } = this.props;
+        if (slug) {
+            this.props.actions.cardFetch(token, slug);
+        }
+    };
```
Comment (Far3t) [inline src/static/containers/AdminCardAdd/index.js]: Extract everything coming from this.props in order to be readable after: 
const { match, token, actions } = this.props;

File: src/static/containers/AdminCardAdd/index.js [hunk 4]
```diff
@@ -57,59 +56,133 @@ class AdminCardAddView extends Component {
...
+            status: PropTypes.string,
+            year: PropTypes.number,
+            data_sources: PropTypes.arrayOf(
+                PropTypes.shape({
+                    name: PropTypes.string,
+                    link: PropTypes.string,
+                    status: PropTypes.string
+                })
+            )
+        }),
```
Comment (Far3t) [inline src/static/containers/AdminCardAdd/index.js]: For PropTypes, you normally don't have to describe the whole object (think about a complex object coming from another lib).
Just the data / keys you use in your component

File: src/static/reducers/cards.js [hunk 5]
```diff
@@ -52,6 +65,48 @@ export default function cardsReducer(state = initialState, action) {
                 isCreatingCard: false
             });
 
+        case CARD_DELETE_REQUEST:
+            return Object.assign({}, state, {
+                isDeletingCard: true
+            });
```
Comment (Far3t) [inline src/static/reducers/cards.js]: return {
  ...state,
  isDeletingCard: true,
}

Comment (Far3t) [reply src/static/reducers/cards.js]: Same for the others! Easier to read:
 - copy the original state
 - update the key isDeletingCard to true

File: src/static/reducers/cards.js [hunk 6]
```diff
@@ -52,6 +65,48 @@ export default function cardsReducer(state = initialState, action) {
...
+
+        case CARD_EDIT_FAILURE:
+            return Object.assign({}, state, {
+                isEditingCard: false
+            });
+
+        case CARD_EDIT_SUCCESS:
+            return Object.assign({}, state, {
+                isEditingCard: false
+            });
```
Comment (Far3t) [inline src/static/reducers/cards.js]: If you do not want any difference between cases, you can group them
But it depends about what you want to do in the future

File: src/static/actions/cards.js (hunk 2 above)
Comment (blatinier) [reply src/static/actions/cards.js]: I squashed these functions in a generic one in the next branch.

File: src/static/actions/cards.js (hunk 1 above)
Comment (blatinier) [reply src/static/actions/cards.js]: Done in the next branch

File: src/static/containers/AdminCardAdd/index.js (hunk 3 above)
Comment (blatinier) [reply src/static/containers/AdminCardAdd/index.js]: OK

File: src/static/reducers/cards.js (hunk 5 above)
Comment (blatinier) [reply src/static/reducers/cards.js]: OK

File: src/static/reducers/cards.js (hunk 6 above)
Comment (blatinier) [reply src/static/reducers/cards.js]: OK

**Pull request closed without merging**